File: Models/Entities/Field.cs
namespace DiplomaThesisDigitalization.Models.Entities
{
    public class Field
    {
        public Field()
        {
            this.Mentors = new HashSet<Mentor>();
        }
        public int Id { get; set; }
        public string FieldName { get; set; }
        public int? DepartmentId { get; set; }
        public Department Department { get; set; }

        public ICollection<Student> Students { get; set; }
        public ICollection<Title> Titles { get; set; }
        public ICollection<Mentor> Mentors { get; set; }
    }
}

File: my-ts-react-app/src/app/models/Field.ts
import { Department } from "./Department";
import { Mentor } from "./Mentor";
import { Student } from "./Student";
import { Title } from "./Title";

export interface Field {
    id: number;
    fieldName: string;
    departmentId: number | null;
    department: Department | null;
    students: Student[];
    titles: Title[];
    mentors: Mentor[];
  }